### ISSUE TITLE
Check for weak signature algorithms

### ISSUE THREAD

**Issue by kroeckx** [ssllabs/ssllabs-scan#740]
Some servers use SHA1 based signature algorithms when the client didn't offer them. See for instance https://github.com/openssl/openssl/issues/7126

Can a check be added for such servers, and stop marking them as A+?

**Comment by sebastianas** [comment]
The key exchange in "forward security" is signed with a hash. The default [security level](https://www.openssl.org/docs/man1.1.1/man3/SSL_CTX_get_security_level.html) can be set to 2 which then eliminates "short" hashes like SHA1. If a server signs the key exchange with SHA1 (which was not offered by the client due to the security level) then the connection will fail.
This is usually a miss configuration / broken software on the server's side.

**Comment by Glandos** [comment]
Alright, but it happens a lot to website that we can't even name and shame. For example, www.gouvernement.fr or bank sites that really don't care about this, since it works in browsers.

Maybe they should be detected by your (really) popular tool.

**Comment by tamthing** [comment]
Thanks for bringing this to our notice, we would investigate based on the evidence and see the changes needed.

**Comment by Darkspirit** [comment]
This can be closed as duplicate of #465.

**Comment by sebastianas** [comment]
Now that we have a [chosen-prefix collision for SHA-1](https://sha-mbles.github.io/), any chance that we can make this sha1 sigantures, that the client did not ask for, more visible?

**Comment by kroeckx** [comment]
Or servers that prefer SHA1 over SHA256 when both are offered by the client,
which is something Windows servers do.

**Comment by rzr** [comment]
It looks like this issue is affecting debian 10, let me crosslink to :

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=934453

I am also able to replicate, if it helps:

```sh
curl 'https://www.gouvernement.fr'
curl: (35) error:1414D172:SSL routines:tls12_check_peer_sigalg:wrong signature type
```


Relate-to: https://twitter.com/RzrFreeFr/status/1242749001964888065

**Comment by Darkspirit** [comment]
Yes, this issue is a duplicate of https://github.com/ssllabs/ssllabs-scan/issues/465#issuecomment-554669595.

**Comment by edmorley** [comment]
Hi! Heroku has recently released its new stack, Heroku-20, which is now based on Ubuntu 20.04 rather than Ubuntu 18.04.

We're getting a number of reports of users who are seeing the openssl `wrong signature type` errors due to Ubuntu 20.04 having changed the OpenSSL default `SECLEVEL` to 2.

It would be great if the SSL Labs test correctly identified these broken server implementations, since currently it reports success, making it a hard sell to try and convince users that it's the third party server that is broken - particularly since these broken implementations still typically work in browsers.

Many thanks :-)

**Comment by derhagen** [comment]
A lot of websites that get an A rating are not viewable on Fedora 33 + Firefox since they introduced their new crypto policies. Several admins ignored my bug report pointing to their good SSLLabs rating.

**Comment by dilyanpalauzov** [comment]
www.euroclear.com is such a site, which according to SSLLabs works under Firefox, but under Fedora/Firefox it does not.
